packages/driver/src/cy/multidomain/index.ts

```diff
@@ -1,41 +1,62 @@
 import Bluebird from 'bluebird'
-import $Log from '../../cypress/log'
 import { createDeferred } from '../../util/deferred'
+import $utils from '../../cypress/utils'
+import $errUtils from '../../cypress/error_utils'
 
-export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy, state: Cypress.State) {
+export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy, state: Cypress.State, config: Cypress.InternalConfig) {
   let timeoutId
 
   // @ts-ignore
-  Cypress.on('cross:domain:html:received', () => {
+  const communicator = Cypress.multiDomainCommunicator
+
+  communicator.on('html:received', () => {
     // when a secondary domain is detected by the proxy, it holds it up
     // to provide time for the spec bridge to be set up. normally, the queue
     // will not continue until the page is stable, but this signals it to go
-    // ahead because we're anticipating multidomain
+    // ahead because we're anticipating multi-domain
     // @ts-ignore
     cy.isAnticipatingMultidomain(true)
 
     // cy.isAnticipatingMultidomain(true) will free the queue to move forward.
     // if the next command isn't switchToDomain, this timeout will hit and
     // the test will fail with a cross-origin error
     timeoutId = setTimeout(() => {
-      // @ts-ignore
       Cypress.backend('ready:for:domain', { shouldInject: false })
     }, 2000)
   })
 
   Commands.addAll({
-    // this isn't fully implemented, but in place to be able to test out
-    // the other parts of multidomain
-    switchToDomain (domain, fn) {
+    switchToDomain<T> (domain: string, dataOrFn: T | (() => {}), fn?: (data?: T) => {}) {
```

Comment:
I'm not sure the types can be deduped since there's different needs between the public types and the implementation. The argument the function takes for the public type isn't optional, but is optional for the implementation.

Eventually, we'll add the other command for multi-domain, `createDomain`, and at that point there will likely be a need to pull out the type, so it will be good to keep in mind.